DevCycle/Utils/RequestConsolidator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  RequestConsolidator.swift
+//  DevCycle
+//
+//  Copyright © 2022 Taplytics. All rights reserved.
+//
+
+import Foundation
+
+class RequestConsolidator {
+    var requestCallbacks: [RequestWithCallback]
+    var service: DevCycleServiceProtocol
+    var requestInFlight: Bool
+    
+    init(service: DevCycleServiceProtocol) {
+        self.service = service
+        self.requestCallbacks = []
+        self.requestInFlight = false
+    }
+    
+    func queue(request: URLRequest, callback: @escaping ConfigCompletionHandler) {
+        if (self.requestInFlight) {
+            self.requestCallbacks.append(
+                RequestWithCallback(
+                    callback: callback,
+                    request: request
+                )
+            )
+            return
+        }
+        
+        self.requestInFlight = true
+        service.makeRequest(request: request) { response in
+            if (self.requestCallbacks.isEmpty) {
+                guard let config = processConfig(response.data) else {
+                    callback((nil, response.error))
+                    return
+                }
+                callback((config, response.error))
+                self.requestInFlight = false
+            } else {
+                self.requestCallbacks.insert(
+                    RequestWithCallback(
+                        callback: callback,
+                        request: request
+                    ),
+                    at: 0
+                )
+                self.makeLastRequestInQueue {
```

Comment:
should this not just call `queue()` again? what happens if another call to `queue()` happens while `makeLastRequestInQueue()` is executing?

Reply:
it will recognize a request would still be in flight and queue it and not make the request.